pac4j-oidc/src/main/java/org/pac4j/oidc/metadata/OidcOpMetadataResolver.java

```diff
@@ -186,4 +194,27 @@ private static ClientAuthenticationMethod firstSupportedMethod(
     protected TokenValidator createTokenValidator() {
         return new TokenValidator(configuration, this.loaded);
     }
+
+    @EqualsAndHashCode(callSuper = true)
+    private static class OidcMetadataUrlResource extends UrlResource {
```

Comment:
Why is it not a public class to allow easy overriding?

Reply:
To be sure, are you asking for `OidcMetadataUrlResource` to be public?

Can do, but I am not sure why anyone would prefer to use an internal class rather than just extending `UrlResource` itself. Also, even if they extend this somehow, where and how could someone use this?

Reply:
Yes, that was my original question. Indeed, people would certainly prefer to extend `UrlResource`...